### PR TITLE
Random Walk Pregel Example

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

I create this pregel example because of an internal request. It will do configurable number of steps along random edges and then store the result at each starting vertex. The paths can contain cycles. Paths can be shorter if there are no outbound edges on a vertex.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

This introduces a new primitive called "list-repeat" to create a list by repeating a given value.

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### CODE REVIEW

**Comment by hkernbach** [comment]
Testsuite needs to be added to:
`tests/js/client/shell/shell-custom-pregel.js`

Otherwise LGTM.

**Comment by maierlars** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/14231/

File: CHANGELOG [hunk 1]
```diff
@@ -15,6 +15,8 @@ devel
 * Added logging of elapsed time of ArangoSearch commit/consolidation/cleanup 
   jobs.
 
+* added list-repeat AIR primitive.
```
Comment (hkernbach) [inline CHANGELOG:18]:
```suggestion
* Added list-repeat AIR primitive.
```

Comment (hkernbach) [reply CHANGELOG:18]: Maybe write add a bit more info here what list-repeat actually does.

Comment (hkernbach) [reply CHANGELOG:18]: Sth. like: e.g. For use in ArangoDB Custom Pregel in order to ...

File: js/client/modules/@arangodb/air/random-walk.js [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+                    "vote-halt"
+                ]
+            }
+        ]
+    };
+}
+
+/* `single_source_shortest_path` executes the program
+   returned by `single_source_shortest_path_program`
+   on the graph identified by `graphName`. */
```
Comment (hkernbach) [inline js/client/modules/@arangodb/air/random-walk.js:308]:
```suggestion

```

**Review by hkernbach — CHANGES_REQUESTED** [review]
Just minor stuff - everything else LGTM

File: CHANGELOG (hunk 1 above)
Comment (maierlars) [reply CHANGELOG:18]:
```suggestion
* Added list-repeat AIR primitive that creates a list containing n copies of the input value.
```

**Review by hkernbach — APPROVED** [review]

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/14334/

**Comment by mchacki** [comment]
Jenkins Blue.

**Pull request merged**